garageScript/c0d3-app
# feat: Make `ExerciseCardPreview` state optional

Related to #2029 
Depends on #2385

### Description

Merging this PR makes it possible to reuse the [ExercisePreviewCard](https://github.com/garageScript/c0d3-app/blob/master/components/ExercisePreviewCard/ExercisePreviewCard.tsx) by making the exercise `state` type optional.

### Testing

- Create a module from `/admin/lessons/js0/modules`
- Create an exercise from `/curriculum/js0/mentor/addExercise`

1. Go to `/curriculum/js0/mentor`
2. Confirm the exercises look like the following

<img width="452" alt="image" src="https://user-images.githubusercontent.com/35906419/193434835-8ac11b5e-4696-4cbb-8526-f2a48110fb5f.png">
